File: leetcode/js/convert-a-number-to-hexadecimal.js
/**
 * @param {number} num
 * @return {string}
 */
var toHex = function (num) {
  const hex = [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
    "a",
    "b",
    "c",
    "d",
    "e",
    "f",
  ];

  if (num === 0) {
    return "0";
  }

  if (num < 0) {
    num += Math.pow(2, 32);
  }

  let result = "";

  while (num > 0) {
    const res = num % 16;
    result = hex[res] + result;
    num = Math.floor(num / 16);
  }

  return result;
};

var toHex = function (num) {
  if (num < 0) {
    return (num >>> 0).toString(16);
  }

  return num.toString(16);
};

File: leetcode/rust/convert-a-number-to-hexadecimal.rs
impl Solution {
    pub fn to_hex(num: i32) -> String {
        if num == 0 {
            return String::from("0");
        }

        let mut num = num as i64;

        if num < 0 {
            num += 2_i64.pow(32);
        }

        let hex = [
            '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f',
        ];
        let mut result = String::new();

        while num > 0 {
            let res = num % 16;
            let hex_num = hex[res as usize];
            result = format!("{hex_num}{result}");
            num = num / 16;
        }

        result
    }
}
